Gems/PhysX/Code/Include/PhysX/EditorColliderComponentRequestBus.h

```diff
@@ -39,85 +39,191 @@ namespace PhysX
+        virtual void SetCylinderRadius(float radius) = 0;
+
+        //! Gets the radius of the cylinder collider.
+        //! @return The radius of the cylinder collider.
+        virtual float GetCylinderRadius() const = 0;
+
+        //! Sets the height of the cylinder collider.
+        //! @param radius The height of the cylinder collider.
+        virtual void SetCylinderHeight(float height) = 0;
+
+        //! Gets the height of the cylinder collider.
+        //! @return The height of the cylinder collider.
+        virtual float GetCylinderHeight() const = 0;
+
+        //! Sets the subdivision count of the cylinder collider.
+        //! @param radius The subdivision count of the cylinder collider.
+        virtual void SetCylinderSubdivisionCount(AZ::u8 subdivisionCount) = 0;
+
+        //! Gets the subdivision count of the cylinder collider.
+        //! @return The subdivision count of the cylinder collider.
+        virtual AZ::u8 GetCylinderSubdivisionCount() const = 0;
+    };
+
+    using EditorPrimitiveColliderComponentRequestBus = AZ::EBus<EditorPrimitiveColliderComponentRequests>;
+
+    //! Request bus for colliders using PhysX mesh assets.
+    class EditorMeshColliderComponentRequests
+        : public AZ::EntityComponentBus
+    {
+    public:
+
+        //! Sets the scale of the asset collider.
+        //! @param The scale of the asset collider.
+        virtual void SetAssetScale(const AZ::Vector3&) = 0;
+
+        //! Gets the scale of the asset collider.
+        //! @return The scale of the asset collider.
+        virtual AZ::Vector3 GetAssetScale() const = 0;
+    };
+
+    using EditorMeshColliderComponentRequestBus = AZ::EBus<EditorMeshColliderComponentRequests>;
+
+    /// O3DE_DEPRECATION_NOTICE(GHI-14717)
     /// <EditorColliderValidationRequests>
     /// This is a Bus in order to communicate the status of the meshes of the collider and avoid dependencies with the rigidbody
     /// </EditorColliderValidationRequests>
-    class EditorColliderValidationRequests : public AZ::ComponentBus
+    class AZ_DEPRECATED(EditorColliderValidationRequests, "Functionality moved to EditorMeshColliderValidationRequests")
+        : public AZ::ComponentBus
     {
     public:
         /// Checks if the the mesh in the collider is correct with the current state of the Rigidbody!
         virtual void ValidateRigidBodyMeshGeometryType() = 0;
     };
 
+    AZ_PUSH_DISABLE_WARNING(4996, "-Wdeprecated-declarations")
     using EditorColliderValidationRequestBus = AZ::EBus<EditorColliderValidationRequests>;
+    AZ_POP_DISABLE_WARNING
+    
+    //! This is a Bus in order to communicate the status of the meshes of the collider and avoid dependencies with the rigidbody
+    class EditorMeshColliderValidationRequests : public AZ::ComponentBus
+    {
+    public:
+        //! Checks if the the mesh in the collider is correct with the current state of the Rigidbody!
+        virtual void ValidateRigidBodyMeshGeometryType() = 0;
```

Comment:
suggestion: This might be outside the scope of this change but it seems like it might be useful if this function returned a `bool` to indicate success or failure of the validation?

Reply:
I understand why you'd make that suggestion, but in this case the caller doesn't need to know. This is just a notification from certain points so the collider runs a validation check on itself to show a message to the user if it's wrongly configured.

Reply:
Okay understood, thanks for the explanation 👍